verification/src/http_server/mod.rs

```diff
@@ -1,22 +1,50 @@
 pub mod handlers;
+pub mod metrics;
 mod routers;
 
 pub use self::routers::{configure_router, AppRouter, Router};
 
 use crate::config::Config;
 use actix_web::{App, HttpServer};
+
+use futures::future;
+use metrics::Metrics;
 use std::sync::Arc;
 
 pub async fn run(config: Config) -> std::io::Result<()> {
     let socket_addr = config.server.addr;
+    let metrics_enabled = config.metrics.enabled;
+    let metrics_addr = config.metrics.addr;
+    let metrics_endpoint = config.metrics.endpoint.clone();
+
     log::info!("Verification server is starting at {}", socket_addr);
     let app_router = Arc::new(
         AppRouter::new(config)
             .await
             .expect("couldn't initialize the app"),
     );
-    HttpServer::new(move || App::new().configure(configure_router(&*app_router)))
+    let metrics = Metrics::new(metrics_endpoint);
+    let server_future = {
+        let middleware = metrics.middleware().clone();
+        HttpServer::new(move || {
+            App::new()
+                .wrap(middleware.clone())
+                .configure(configure_router(&*app_router))
+        })
         .bind(socket_addr)?
         .run()
-        .await
+    };
+    let server_future = tokio::spawn(async move { server_future.await });
+    let metrics_future = tokio::spawn(async move {
+        if metrics_enabled {
```

Comment:
I guess it's easier to check if here, but I'd prefer to actually do nothing, if metrics not enabled, including not spawning a thread. You can do this using vec
```rust
let futures = vec![tokio::spawn(server_future...)];
if metrics_enabled {
  futures.push(tokio::spawn(metrics...));
}
```